paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.paimon.spark
+
+import org.apache.paimon.data.BinaryRow
+import org.apache.paimon.operation.FileStoreCommit
+import org.apache.paimon.table.{AbstractFileStoreTable, Table}
+import org.apache.paimon.table.sink.BatchWriteBuilder
+import org.apache.paimon.table.source.TableScan
+import org.apache.paimon.types.RowType
+import org.apache.paimon.utils.{FileStorePathFactory, RowDataPartitionComputer}
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.{CatalystTypeConverters, InternalRow}
+import org.apache.spark.sql.connector.catalog.SupportsPartitionManagement
+import org.apache.spark.sql.types.StructType
+
+import java.util
+import java.util.{Collections, UUID}
+
+import scala.collection.JavaConverters._
+
+abstract class PaimonPartitionManagement(table: Table) extends SupportsPartitionManagement {
+
+  override def partitionSchema(): StructType = {
+    val partitionKeys = table.partitionKeys
+    val rowType = new RowType(
+      table.rowType.getFields.asScala
+        .filter(dataFiled => partitionKeys.contains(dataFiled.name()))
+        .asJava)
+    SparkTypeUtils.fromPaimonRowType(rowType)
+  }
+
+  override def dropPartition(internalRow: InternalRow): Boolean = {
+    val partitionKeys = table.partitionKeys
+    val rowType = new RowType(
+      table.rowType.getFields.asScala
+        .filter(dataFiled => partitionKeys.contains(dataFiled.name()))
+        .asJava)
+    // convert internalRow to row
+    val row: Row = CatalystTypeConverters
+      .createToScalaConverter(partitionSchema())
+      .apply(internalRow)
+      .asInstanceOf[Row]
+    val rowDataPartitionComputer = new RowDataPartitionComputer(
+      FileStorePathFactory.PARTITION_DEFAULT_NAME.defaultValue,
+      rowType,
+      partitionKeys.asScala.toArray)
+    val partitionMap = rowDataPartitionComputer.generatePartValues(new SparkRow(rowType, row))
+    val identifier: Long = BatchWriteBuilder.COMMIT_IDENTIFIER
+    val commit: FileStoreCommit =
+      (table.asInstanceOf[AbstractFileStoreTable]).store.newCommit(UUID.randomUUID.toString)
```

Comment:
`FileStoreCommit newCommit(String commitUser);`

here why use UUID to fake the commit user?

Reply:
because i saw `((AbstractFileStoreTable) table).store().newCommit(UUID.randomUUID().toString());` in other modules.